ddtrace/propagation/http.py

```diff
@@ -130,6 +178,40 @@ def _extract(headers):
             headers,
         )
 
+        meta = None
+        tags_value = _extract_header_value(
+            POSSIBLE_HTTP_HEADER_TAGS,
+            headers,
+            default="",
+        )
+        if tags_value:
+            # Do not fail if the tags are malformed
+            try:
+                # We get a Dict[str, str], but need it to be Dict[Union[str, bytes], str] (e.g. _MetaDictType)
+                meta = cast(
+                    Dict[Union[str, bytes], str],
+                    {
+                        k: v
+                        for (k, v) in decode_tagset_string(tags_value).items()
+                        if k not in _DatadogMultiHeader._X_DATADOG_TAGS_REJECT
+                    },
+                )
+            except TagsetMaxSizeDecodeError:
+                meta = {
+                    "_dd.propagation_error": "extract_max_size",
+                }
+                log.warning("failed to decode x-datadog-tags", exc_info=True)
+            except TagsetDecodeError:
+                meta = {
+                    "_dd.propagation_error": "decoding_error",
+                }
+                log.debug("failed to decode x-datadog-tags: %r", tags_value, exc_info=True)
+            finally:
+                # Datadog propagation should still attempt to decode even if tag
+                # propagation is disabled
+                if not config._x_datadog_tags_enabled:
+                    meta = None
```

Comment:
We don't want to replace `meta` if an exception has been raised, no need for cleanup.

```suggestion
```